src/coreclr/src/jit/codegencommon.cpp

```diff
@@ -6338,71 +6312,237 @@ void CodeGen::genZeroInitFrame(int untrLclHi, int untrLclLo, regNumber initReg,
         noway_assert(uCntBytes == 0);
 
 #elif defined(TARGET_XARCH)
-        /*
-            Generate the following code:
-
-                lea     edi, [ebp/esp-OFFS]
-                mov     ecx, <size>
-                xor     eax, eax
-                rep     stosd
-         */
+        assert(compiler->getSIMDSupportLevel() >= SIMD_SSE2_Supported);
+        emitter*  emit        = GetEmitter();
+        regNumber frameReg    = genFramePointerReg();
+        regNumber zeroReg     = REG_NA;
+        int       blkSize     = untrLclHi - untrLclLo;
+        int       minSimdSize = XMM_REGSIZE_BYTES;
+
+        assert(blkSize >= 0);
+        noway_assert((blkSize % sizeof(int)) == 0);
+        // initReg is not a live incoming argument reg
+        assert((genRegMask(initReg) & intRegState.rsCalleeRegArgMaskLiveIn) == 0);
+#if defined(TARGET_AMD64)
+        // We will align on x64 so can use the aligned mov
 
-        noway_assert(regSet.rsRegsModified(RBM_EDI));
+        // We use movaps at R2R because it is a smaller instruction; however at JIT time
+        // the VEX version vmovaps would be used which is the same size as vmovdqa;
+        // also vmovdqa has more available CPU ports on older processors so we switch to that
+        instruction simdMov = compiler->opts.IsReadyToRun() ? INS_movaps : INS_movdqa;
```

Comment:
IsReadyToRun does not look like the right condition to use to check whether the JIT is emitting VEX instructions. Is there a better way to check whether the JIT is emitting VEX instructions?

Reply:
> IsReadyToRun does not look like the right condition to use to check whether the JIT is emitting VEX instructions. Is there a better way to check whether the JIT is emitting VEX instructions?

@tannergooding advice?

Reply:
`compiler->canUseVexEncoding()` looks like

Reply:
Resolved

Reply:
Yes `Compiler::canUseVexEncoding()` is what you want. Also, sorry for the delay, I'm in London this week and didn't have immediate access to a computer.